### ISSUE TITLE
add command to allow switching to visual mode from other addons

### ISSUE THREAD

**Issue by david-morris** [VSCodeVim/Vim#9213]
**Problem**

I use [Owlbear ](https://github.com/sansarip/owlbear)and I'd like to have the option of dropping into visual mode from the new "select node" command.
That means I would need a command that I could set a keybinding for that would let me go from VSCode selection to `VISUAL` selection.

**Proposed solution**

Add a command to the package.json that triggered a function that used the VSCode selection as a new visual mode selection.

**Alternatives**

The setting to go into visual mode on mouse select could be modified to listen to selections in general, but that would limit users' ability to choose when they went into visual mode and could interfere with snippet workflows.

**Additional context**

In general, I think it would be nice to expose stuff for other addons to do things like set modes, register commands as having been performed for the purpose of macros, save/restore macros, etc.

**Comment by HenryTSZ** [comment]
@david-morris I use `Owlbear: Select Current Node` and the vim entered visual mode

**Comment by HenryTSZ** [comment]
@david-morris I use `Owlbear: Select Current Node` and the vim entered visual mode

**Comment by david-morris** [comment]
Hmm, that didn't work when I posted it!  Visual keybinds now work when I do that.

Weird thing though, I only get `--VISUAL--` in my modeline when I started in insert mode.  If I started in normal mode, nothing shows up in my modeline.

I'm closing this as not planned because the specific example is obsolete, and this issue is too specific to really start a conversation about larger integration with e.g. macros including commands.
